pkg/apis/core/validation/validation.go

```diff
@@ -4064,8 +4065,12 @@ func ValidateService(service *core.Service) field.ErrorList {
 		}
 	}
 
-	allErrs = append(allErrs, validateServiceExternalTrafficFieldsValue(service)...)
+	//if an ipfamily provided then it has to be one of the supported values
+	if service.Spec.IPFamily != nil && !supportedServiceIPFamily.Has(string(*service.Spec.IPFamily)) {
```

Comment:
Do we need to accept nil here?  Validation should be called AFTER the default value is set.  If dual-stack is gated off, it will be nil here, but we should not be validating it at all then.

Reply:
the validation accepts nil, specifically for the reasons you outlined. We also validate if there is a value then it needs to match what is allowed.